src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -516,6 +516,14 @@ public void AddFormatter(ValueFormatterFactory formatterFactory)
             CurrentValueFormatter = formatterFactory(CurrentValueFormatter);
         }
 
+        /// <summary>
+        /// Clear reference to test object
+        /// </summary>
+        internal void ClearTestObject()
+        {
+            TestObject = null;
+        }
+
```

Comment:
Why add a method, given that TestObject has a public setter, which we use elsewhere?